src/js/client/editprefset-dialogs-widget.js

```diff
@@ -0,0 +1,348 @@
+    }
+    // TODO standardize the accepted regular expression for prefset ID's
+    // and other parts of the schema.
+    else if (!/^[\w-]+$/.exec(model.contextId)) {
+        applier.change("contextIdErrors", "Please use only alphanumeric characters.");
+        return;
+    }
+    // If we changed the contextId, make sure that there isn't an existing context with
+    // that ID
+    else if (model.contextId !== model.originalContextId &&
+             gpii.devpmt.prefsetExists(model.flatPrefs, model.contextId)) {
+        applier.change("contextIdErrors", "A Preference Set with that ID already exists.");
+        return;
+    }
+    else {
+        // In case we are revalidating on a subsequent attempt
+        applier.change("contextIdErrors", "");
+    }
+
+    if (model.contextName === "") {
+        applier.change("contextNameErrors", "Please enter a name.");
+        return;
+    }
+    else {
+        // In case we are revalidating on a subsequent attempt
+        applier.change("contextNameErrors", "");
+    }
+
+    var transaction = applier.initiate();
+    var prefsetToEdit = fluid.copy(model.flatPrefs.contexts[model.originalContextId]);
+    prefsetToEdit.name = model.contextName;
+    var oldPath = "flatPrefs.contexts." + model.originalContextId;
+    var path = "flatPrefs.contexts." + model.contextId;
+
+    transaction.fireChangeRequest({
+        path: oldPath,
+        type: "DELETE"
+    });
+
+    transaction.fireChangeRequest({
+        path: path,
+        value: prefsetToEdit
+    });
+
+    transaction.commit();
+    closeDialog();
+};
+
+
+/**
+ * Confirm deletion of Preferences Set Context Dialog
+ */
+fluid.defaults("gpii.devpmt.dialogs.confirmDeleteContextDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.confirmDialog"],
+    templates: {
+        initial: "editprefset-confirmDeleteContext-dialog"
+    },
+    model: {
+        contextId: "", // Should be populated/relayed during construction before showing dialog
+        flatPrefs: {}
+    },
+    invokers: {
+        acceptConfirmDialog: {
+            funcName: "gpii.devpmt.dialogs.confirmDeleteContextDialog.acceptConfirmDialog",
+            args: ["{that}.closeDialog", "{gpii.devpmt.editPrefs}", "{that}.model.contextId"]
+        }
+    }
+});
+
+gpii.devpmt.dialogs.confirmDeleteContextDialog.acceptConfirmDialog = function (closeDialog, editPrefs, contextId) {
+    closeDialog();
+    // In the rare event that this contextId no longer exists in the preferences,
+    // the worse case scenerio here is that the change applier operation will merely
+    // do nothing.
+    var path = "flatPrefs.contexts." + contextId;
+    //TODO We want to use the local change applier for this operation and have the model Relay
+    //push the DELETE back to the editPrefs component. However, the DELETE does not seem to be
+    //propagating. Investigate more.
+    editPrefs.applier.change(path, false, "DELETE");
+};
+
+/**
+ * Add Product to Preferences Set Dialog
```

Comment:
Why do we refer to a "product" rather than a "solution"?

Reply:
Hmm... I think we just started referring to them as products in the context of the PPT wireframes and review meetings, and I just started using it this scenerio for the dialogs...